sunpy/net/dataretriever/tests/test_swap.py

```diff
@@ -0,0 +1,58 @@
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+import datetime
+import pytest
+
+from sunpy.time.timerange import TimeRange
+from sunpy.net.vso.attrs import Time,Instrument,Level
+from sunpy.net.dataretriever.client import QueryResponse
+from sunpy.net.dataretriever.downloader_factory import UnifiedResponse
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+import sunpy.net.dataretriever.sources.swap as swap
+
+LCClient = swap.SWAPClient()
+
+@pytest.mark.online
+@pytest.mark.parametrize("timerange,url_start,url_end, level",
+                         [(TimeRange('2015/12/30 00:00:00','2015/12/30 23:59:59'),
+                           'http://proba2.oma.be/swap/data/bsd/2015/12/30/swap_lv1_20151230_000044.fits',
+                           'http://proba2.oma.be/swap/data/bsd/2015/12/30/swap_lv1_20151230_235935.fits', 1)])
+def test_get_url_for_time_range(timerange, url_start, url_end, level):
+    urls = LCClient._get_url_for_timerange(timerange, level = level)
+    assert isinstance(urls, list)
+    assert urls[0] == url_start
+    assert urls[-1] == url_end
+
+def test_can_handle_query():
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap'), a.Level(1)) == True
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap'), a.Level('q')) == True
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap'), a.Level('s')) == False
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap')) == False
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30','2015/12/31')) == False
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('eve')) == False
```

Comment:
@wafels I'm not sure what's explicit would be here... if I have a function that does:
```python
def workday(dayofweek)
    if dayofweek is in ('Saturday', 'Sunday'):
        return False
    return True
```
then the test could be written as:
```python
assert workday('Tuesday')
assert not workday('Sunday')
```
and regarding the `is` vs `==` I think the best example is in here:
```python
>>> assert 1 == True
>>> assert 1 is True
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-20-3de0087e2891> in <module>()
----> 1 assert 1 is True

AssertionError: 
```
so, using `is` will fail if the signature of the function changes! which is what we want to catch.